### PR TITLE
15.0 3258685 mhch

### PR DESCRIPTION
**Pull request by mhch-odoo**
### Description

*Please include a summary of the change, the project task link and/or the ID.*

Link to task: [#3258685](https://www.odoo.com/web#model=project.task&id=3258685)

### All Submissions:

* [ ] My commit respects the [Odoo commit guideline](https://www.odoo.com/documentation/15.0/developer/misc/other/guidelines.html#git)
* [ ] My commit message respects the [commit template](https://github.com/odoo-ps/psbe-process/wiki/Commits-message-guidelines#template)
* [ ] I have used pre-commit
* [ ] The PR contains **only** my modification and **no other external** commit

### Sh/Runbot:

* [ ] The commits pass test and the branch is green
* [ ] Unit tests have been implemented / standard ones rewritten
* [ ] The Staging is ISO-Prod and will contain only this dev

### Upgrade:

* [ ] The data affected (*if any*) by the changes has been migrated 

### Maintenance:

* [ ] The versions of the modules affected by the changes have been bumped in their manifests

### CODE REVIEW

File: sakaya_zonos/static/src/js/checkout.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        /**
+         * @override
+         */
+        _handleCarrierUpdateResult: function (result) {
+            this._super.apply(this, arguments);
+            var $amountDeliveryTax = $('#order_delivery_tax .monetary_field');
+            var $amountShipping = $('#order_shipping .monetary_field');
+            $amountDeliveryTax.html(result.new_amount_delivery_tax);
+            $amountShipping.html(result.new_amount_shipping);
+
```
Comment (fja-odoo) [inline sakaya_zonos/static/src/js/checkout.js:18]: extra space

File: sakaya_zonos/static/src/js/checkout.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        },
+
+    });
+    });
+
```
Comment (fja-odoo) [inline sakaya_zonos/static/src/js/checkout.js:23]: end of file line

File: sakaya_zonos/static/src/js/checkout.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+odoo.define('sakaya_zonos.checkout', function (require) {
+    'use strict';
+    
+    var publicWidget = require('web.public.widget');
```
Comment (fja-odoo) [inline sakaya_zonos/static/src/js/checkout.js:4]: should use const and not var, same below

File: sakaya_zonos/models/stock_picking.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+    def send_dropship_confirmation_email(self):
+        # Send confirmation to vendor
+        delivery_vendor_template = self.company_id.stock_mail_confirmation_template_id
+        label = self.env["ir.attachment"].search([("res_id", "=", self.id), ("res_model", "=", "stock.picking"), ("name", 'like', self.carrier_tracking_ref)])
+        if label:
+            email_values = {'attachment_ids': [(4, label.id)]}
+        else:
+            raise UserError("No Shipping Label Found!")
+        delivery_vendor_template.send_mail(self.id, email_values=email_values, notif_layout="mail.mail_notification_light", force_send=True)
```
Comment (fja-odoo) [inline sakaya_zonos/models/stock_picking.py:22]: don't do `force_send=True` best to let the cron send them in batch

File: sakaya_zonos/models/sale_order_line.py [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+from odoo import models, fields
+from collections import defaultdict
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    merchant_id = fields.Many2one("res.partner", related="product_id.merchant_id")
+
+    def group_by_merchants(self):
```
Comment (fja-odoo) [inline sakaya_zonos/models/sale_order_line.py:11]: It is a bit weird to create a method for something only used once

**Review by fja-odoo — COMMENTED** [review]
First few comments

File: sakaya_zonos/models/sale_order_line.py (hunk 5 above)
Comment (mhch-odoo) [reply sakaya_zonos/models/sale_order_line.py]: It's to be used in the views as well

**Pull request closed without merging**